File: src/main/java/no/nav/service/pensjon/regler/to/RegulerBeregning2011Request.java
package no.nav.service.pensjon.regler.to;

import java.util.ArrayList;
import java.util.Date;

import no.nav.domain.pensjon.regler.beregning2011.AbstraktBeregningsResultat;
import no.nav.domain.pensjon.regler.beregning2011.AfpLivsvarig;
import no.nav.domain.pensjon.regler.grunnlag.DelingstallUtvalg;
import no.nav.domain.pensjon.regler.grunnlag.ForholdstallUtvalg;
import no.nav.domain.pensjon.regler.grunnlag.Uttaksgrad;

public class RegulerBeregning2011Request extends ServiceRequest {
	private Date virkFom;
	private AbstraktBeregningsResultat beregningsResultat;
	private AfpLivsvarig afpLivsvarig;
	private ArrayList<Uttaksgrad> uttaksgradListe;
	private Date fodselsdato;
	private ForholdstallUtvalg forholdstallUtvalg;
	private DelingstallUtvalg delingstallUtvalg;
}

File: src/main/kotlin/no/nav/pensjon/regler/to/RegulerBeregning2011Request.kt
package no.nav.pensjon.regler.to

import no.nav.pensjon.regler.domain.beregning2011.AbstraktBeregningsResultat
import no.nav.pensjon.regler.domain.beregning2011.AfpOffentligLivsvarigGrunnlag
import no.nav.pensjon.regler.domain.beregning2011.AfpPrivatLivsvarig
import no.nav.pensjon.regler.domain.grunnlag.NormertPensjonsalderGrunnlag
import no.nav.pensjon.regler.domain.grunnlag.Uttaksgrad
import java.util.*
import java.time.LocalDate

class RegulerBeregning2011Request : ServiceRequest() {
    @Deprecated("Use virkFomLd instead")
    var virkFom: Date? = null
    var virkFomLd: LocalDate? = null
    var beregningsResultat: AbstraktBeregningsResultat? = null
    var afpPrivatLivsvarig: AfpPrivatLivsvarig? = null
    var uttaksgradListe: ArrayList<Uttaksgrad> = ArrayList()
    @Deprecated("Use fodselsdatoLd instead")
    var fodselsdato: Date? = null
    var fodselsdatoLd: LocalDate? = null

    @Deprecated("Felt ikke i bruk. AfpOffentligLivsvarig er forventet satt på beregningsResultat")
    var afpOffentligLivsvarigGrunnlag: AfpOffentligLivsvarigGrunnlag? = null

    /**
     * Representerer grunnlaget for normert pensjonsalder
     *
     * Dette feltet inneholder normert, øvre og nedre pensjonsalder (i år og måneder)
     * som benyttes i saksbehandlingen.
     */
    var normertPensjonsalderGrunnlag: NormertPensjonsalderGrunnlag? = null
}